main/src/library/Result.flix

```diff
@@ -0,0 +1,202 @@
+        case (Err(w1), _) => r1
+        case (_, Err(w2)) => r2
+        case (Ok(v1), Ok(v2)) => f(v1, v2)
+    }
+
+    ///
+    /// Returns `r2` if `r1` is `Ok(v)`. Otherwise returns `r1`.
+    ///
+    def and[t, e](r1: Result[t, e], r2: Result[t, e]): Result[t, e] = match r1 with {
+        case Ok(v) => r2
+        case Err(w) => r1
+    }
+
+    ///
+    /// Returns `r1` if `r1` is `Ok(v)`. Otherwise returns `r2`.
+    ///
+    def or[t, e](r1: Result[t, e], r2: Result[t, e]): Result[t, e] = match r1 with {
+        case Ok(v) => r1
+        case Err(w) => r2
+    }
+
+    ///
+    /// Returns `1` if `r` is `Ok(v)` and the predicate `f(v)` evaluates to `true`. Otherwise returns `0`.
+    ///
+    def count[t, e](f: t -> Bool, r: Result[t, e]): Int32 = match r with {
+        case Ok(v) => if (f(v)) 1 else 0
+        case Err(w) => 0
+    }
+
+    ///
+    /// Returns `r` if `r` is `Ok(v)` and the predicate `f(v)` evaluates to `true`. Otherwise returns `Err(d)`.
+    ///
+    def find[t, e](f: t -> Bool, r: Result[t, e], d: e): Result[t, e] = match r with {
+        case Ok(v) => if (f(v)) r else Err(d)
+        case Err(w) => Err(d)
+    }
+
+    ///
+    /// Returns `f(z, v)` if `r` is `Ok(v)`. Otherwise returns `z`.
+    ///
+    def foldLeft[t, e, a](f: (a, t) -> a, z: a, r: Result[t, e]): a = match r with {
+        case Ok(v) => f(z, v)
+        case Err(w) => z
+    }
+
+    ///
+    /// Returns `f(v, z)` if `r` is `Ok(v)`. Otherwise returns `z`.
+    ///
+    def foldRight[t, e, a](f: (t, a) -> a, r: Result[t, e], z: a): a = match r with {
+        case Ok(v) => f(v, z)
+        case Err(w) => z
+    }
+
+    ///
+    /// Returns a one-element list of the value `v` if `r` is `Ok(v)`. Otherwise returns the empty list.
+    ///
+    def toList[t, e](r: Result[t, e]): List[t] = match r with {
+        case Ok(v) => v :: Nil
+        case Err(w) => Nil
+    }
+
+    ///
+    /// Returns a one-element set of the value `v` if `r` is `Ok(v)`. Otherwise returns the empty set.
+    ///
+    def toSet[t, e](r: Result[t, e]): Set[t] = match r with {
+        case Ok(v) => #{v}
+        case Err(w) => #{}
+    }
+
+    ///
+    /// Returns `Some(v)` if `r` is `Ok(v)`. Otherwise returns `None`.
+    ///
+    def toOption[t, e](r: Result[t, e]): Option[t] = match r with {
+        case Ok(v) => Some(v)
+        case Err(w) => None
+    }
+
+    ///
+    /// Returns `Ok((v1, v2))` if `r1` is `Ok(v1)` and `r2` is `Ok(v2)`.
+    /// Otherwise returns the first of `r1` and `r2` that is `Err(w)`.
+    ///
+//    def zip[t1, t2, e](r1: Result[t1, e], r2: Result[t2, e]): Result[(t1, t2), e] = match (r1, r2) with {
+//        case (Err(w1), _) => r1
+//        case (_, Err(w2)) => r2
+//        case (Ok(v1), Ok(v2)) => Ok((v1, v2))
+//    }
+
+    ///
+    /// Returns `(Ok(v1), Ok(v2))` if `r` is `Ok((v1, v2))`. Otherwise returns `(r, r)`.
+    ///
+//    def unzip[t1, t2, e](r: Result[(t1, t2), e]): (Result[t1, e], Result[t2, e]) = match r with {
```

Comment:
Actually unzip makes sense, but let's remove it for consistency, now that we remove zip.